### PR TITLE
session: fix upgrade

### LINKED ISSUE #41502: Some upgrade steps are missed
## Bug Report

Please answer these questions before submitting your issue. Thanks!
1. version109 is tidb_enable_gc_aware_memory_track at v6.5.1
version109 is stats_meta_history and version110 is tidb_enable_gc_aware_memory_track at v6.6 and master.
When upgrade from v6.5.1 to v6.6, upgrade for stats_meta_history will not be executed

2. There are no version numbers to use if we want to add more upgrade steps for v6.5.x

### 1. Minimal reproduce step (Required)
upgrade from v6.5.1 to v6.6

### 2. What did you expect to see? (Required)
Success

### 3. What did you see instead (Required)
Some upgrade steps are missed.

### 4. What is your TiDB version? (Required)
v6.5.1, v6.6, master

### PR DESCRIPTION
**Pull request by chrysan**
### What problem does this PR solve?

Issue Number: close #41502 

Problem Summary:
It's because https://github.com/pingcap/tidb/pull/40179/files. To make it work when upgrade from v6.5 to master, change the upgrade version.
And leave some version numbers for possible patches for v6.5.x

### Check List

Tests 

- [ ] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
None
```

### CODE REVIEW

**Review by winoros — APPROVED** [review]

**Review by time-and-fate — APPROVED** [review]

**Review by XuHuaiyu — APPROVED** [review]

**Review by XuHuaiyu — APPROVED** [review]

**Review by bb7133 — APPROVED** [review]
LGTM

**Review by XuHuaiyu — APPROVED** [review]

File: server/tidb_library_test.go [hunk 1]
```diff
@@ -49,7 +49,7 @@ func TestMemoryLeak(t *testing.T) {
 	runtime.ReadMemStats(&memStat)
 	// before the fix, initAndCloseTiDB for 20 times will cost 900 MB memory, so we test for a quite loose upper bound.
 	if syncutil.EnableDeadlock {
-		require.Less(t, memStat.HeapInuse-oldHeapInUse, uint64(1100*units.MiB))
+		require.Less(t, memStat.HeapInuse-oldHeapInUse, uint64(1500*units.MiB))
```
Comment (chrysan) [inline server/tidb_library_test.go:52]: Tests failed several times. Discussed with @hawkingrei to increase this value now and find a better solution later. 
```
Error:              "1158316032" is not less than "1153433600"
Error:              "1162543104" is not less than "1153433600"
```

**Pull request merged**